### PR TITLE
Deprecate community.azure

### PR DESCRIPTION
**Pull request by mariolenz**
Deprecate community.azure as discussed in ansible-community/community-topics#263 and voted on in ansible-community/community-topics#264

### CODE REVIEW

**Review by felixfontein — APPROVED** [review]

File: 8/changelog.yaml [hunk 1]
```diff
@@ -141,3 +141,11 @@ releases:
 
         `Porting Guide <https://docs.ansible.com/ansible/devel/porting_guides.html>`_'
     release_date: '2023-08-15'
+  8.4.0:
+    changes:
+      deprecated_features:
+      - The community.azure collection is officially unmaintained and has been archived.
```
Comment (gotmax23) [inline 8/changelog.yaml:147]:
```suggestion
      - The ``community.azure`` collection is officially unmaintained and has been archived.
```

File: 9/changelog.yaml [hunk 2]
```diff
@@ -50,3 +50,8 @@ releases:
           from Ansible 10 if no one starts maintaining it again before Ansible 10. See
           `the removal process for details on how this works <https://github.com/ansible-collections/overview/blob/main/removal_from_ansible.rst#cancelling-removal-of-an-unmaintained-collection>`__
           (https://github.com/ansible-community/community-topics/issues/244).
+        - The community.azure collection is officially unmaintained and has been archived.
```
Comment (gotmax23) [inline 9/changelog.yaml:53]:
```suggestion
        - The ``community.azure`` collection is officially unmaintained and has been archived.
```

**Comment by mariolenz** [comment]
@gotmax23 I've merged your suggestions. Anything else you would like to have changed, or is the PR OK now?

**Review by gotmax23 — APPROVED** [review]
LGTM now. Thanks!

**Pull request merged**